src/IO/WriteBufferFromS3.cpp

```diff
@@ -164,34 +170,82 @@ void WriteBufferFromS3::writePart()
         LOG_WARNING(log, "Maximum part number in S3 protocol has reached (too many parts). Server may not accept this whole upload.");
     }
 
+    auto part_data = std::make_shared<Aws::StringStream>();
+    temporary_buffer.swap(part_data);
```

Comment:
What is temporary buffer

Reply:
`WriteBufferFromS3` saves data to temporary buffer until it decides to send it to server. This is `StringStream` because AWS SDK allows to pass body as `StringStream`.